index.js

```diff
@@ -45,28 +69,88 @@ exports.main = function(){
 	simplePrefs.on('maxPrivateBrowsing', function(){
 		maxPrivateBrowsing = simplePrefs.prefs.maxPrivateBrowsing;
 	});
+    simplePrefs.on('maxOpenInNextWindow', function(){
+        maxOpenInNextWindow = simplePrefs.prefs.maxOpenInNextWindow;
+    });
+    simplePrefs.on('maxExcludePinnedTabs', function(){
+        maxExcludePinnedTabs = simplePrefs.prefs.maxExcludePinnedTabs;
+        updateAllButtons();
+    });
 
-	tabs.on('open', function(tab){
-		var window = tab.window;
-		// setTimeout is needed because window.tabs.length value seems to update *slower*
-		setTimeout(function(){
-			var tabsLen = window.tabs.length;
-			if (tabsLen <= max || (privateBrowsing.isPrivate(window) && !maxPrivateBrowsing)){
-				updateButton(window, tabsLen);
-			} else {
-				tab.close();
-				notifications.notify({
-					title: title,
-					text: _('not_open_max_tabs', max)
-				});
-			}
-		}, 1);
-	});
+
+    // TODO: hook tab (un)pinned events
+
+
+    tabs.on('open', function(tab){
+        let win = tab.window;
+        if(privateBrowsing.isPrivate(win) && !maxPrivateBrowsing){
+            setTimeout(function(){
+                updateButton(win, calcTabsLen(win));
+            }, 1);
+            return;
+        }
```

Comment:
As stated [here](https://github.com/rindeal/max-tabs/blob/master/index.js#L104), private windows are always ignored (I'm not even quite sure how to list private windows). This part just makes sure, that if you disabled this addon for private windows in its preferences, it really won't be blocking you in them and will just update its counters.